File: B-Regressions/RegressionTemplates/RegressionTemplate.r
#Regression template

#======================================== Preprocessing Phase ========================================================
dataset = read.csv('./B-Regressions/PolinomialLinearRegression/data/Position_Salaries.csv')
dataset = dataset[2:3]

#Splitting the dataset into Training set and Test set
# install.packages('caTools')
# library('caTools')
# set.seed(123)#Forces same results
# split = sample.split(dataset$Purchased, SplitRatio = 0.8) #Splits data into categories true and false
# training_set = subset(dataset, split == TRUE) #recieves the subset of datasat containing true
# test_set = subset(dataset, split == FALSE) #recieves the subset of datasat containing false

#Feature Scaling
#The first and last columns arent numeric so we need to scale only numbers
# training_set[, 2:3] = scale(training_set[, 2:3])#Scales only columns 2 and 3
# test_set[, 2:3] = scale(test_set[, 2:3])#Scales only columns 2 and 3


#======================================== Creating the model ========================================================
#Fitting the polynomial regression to the dataset
#Create your regressor here

#Predicting a new result
y_pred = predict(regressor, data.frame(Level = 6.5, Level2 = 6.5^2, Level3 = 6.5^3, Level4 = 6.5^4))
print(y_pred)

#Visualizing the regression model Results
#install.packages('ggplot2')
library('ggplot2')
ggplot() +
    geom_point(aes(x = dataset$Level, y = dataset$Salary), color = 'red') +
    geom_line(aes(x = dataset$Level, y = predict(regressor, newdata = dataset)), color = 'blue') +
    ggtitle('Truth or Bluff (Linear Regression)') + 
    xlab('Level') + 
    ylab('Salary')

#Visualizing the regression model (with higher resolution)
#install.packages('ggplot2')
library('ggplot2')
x_grid = seq(min(dataset$Level),max(dataset$Level), 0.1)
ggplot() +
    geom_point(aes(x = dataset$Level, y = dataset$Salary), color = 'red') +
    geom_line(aes(x = x_grid$Level, y = predict(regressor, newdata = data.frame(Level = x_grid))), color = 'blue') +
    ggtitle('Truth or Bluff (Linear Regression)') + 
    xlab('Level') + 
    ylab('Salary')

File: B-Regressions/RegressionTemplates/RegressionTemplate.py
#Polinomial regression

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Import dataset and store it in two separate vectors
dataset = pd.read_csv('./B-Regressions/PolinomialLinearRegression/data/Position_Salaries.csv')
x = dataset.iloc[:, 1:2].values #Matrix of features - Always treat the feature matrix as a matrix doing 1:2 and not just 1
y = dataset.iloc[:, 2].values #Dependent variables set

#Split the dataset into Training set and a Test set
"""from sklearn.cross_validation import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2, random_state = 0)"""

#Feature scaling
"""from sklearn.preprocessing import StandardScaler
sc_x = StandardScaler()
x_train = sc_x.fit_transform(x_train)
x_test = sc_x.transform(x_test)
print(x_train)
print(x_test)"""

#Fitting regression model to the dataset
#Create regressor

#Predicting a new result with Polinomial Regression
y_pred = regressor.predict(6.5)

#Visualizing the polinomial regression results
plt.scatter(x, y, color = 'red')
plt.plot(x, regressor.predict(x), color = 'blue')
plt.title('Truth or Bluff (Polinomial regression)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()

#Visualizing the polinomial regression results with higher resolution
x_grid = np.arange(min(x), max(x), 0.1)
x_grid = x_grid.reshape((len(x_grid), 1))#Reshapes the whole x plot to avoit straight lines between points and increase precison

plt.scatter(x, y, color = 'red')
plt.plot(x, regressor.predict(x), color = 'blue')
plt.title('Truth or Bluff (Polinomial regression)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()